js/index.js

```diff
@@ -30,4 +28,46 @@ for (let i = 0; i < skills.length; i++) {
   skillsList.appendChild(skill);
 }
 
+/**Handle Message Form Submit */
+const messageForm = document.getElementById("leave_message");
+const handleSubmit = (event) => {
+  event.preventDefault();
+
+  let usersName = event.target.usersName.value;
+  let usersEmail = event.target.usersEmail.value;
+  let usersMessage = event.target.usersMessage.value;
+
+  console.log(usersName);
+  console.log(usersEmail);
+  console.log(usersMessage);
+
+  const messageSection = document.getElementById("messages");
+  const messageList = messageSection.querySelector("ul");
+  const newMessage = document.createElement("li");
+
+  newMessage.innerHTML = `
+  <a href="mailto:${usersEmail}">${usersName}</a>
+  <span>: ${usersMessage}</span>
+`;
+
+  const removeButton = document.createElement("button");
+  removeButton.textContent = "Remove";
+  removeButton.setAttribute("type", "button");
+
+  removeButton.addEventListener("click", (event) => {
+    const entry = removeButton.parentNode;
+    entry.remove();
+  });
+
+  newMessage.appendChild(removeButton);
+  messageList.appendChild(newMessage);
+
+  if (messageList.children.length === 0) {
+    messageSection.style.display = "none";
```

Comment:
You're on the right track with managing the visibility! To take it a step further, I recommend separating the logic for showing the section into its own function, like `showMessages`, which you will be calling multiple times.

In `showMessages` function, you can build on your current idea by also displaying the section when the length is greater than 0.

This function should be called when:
- the page is loaded (using a `DOMContentLoaded` listener)
- a 'Remove' button is clicked (which you’ve already set up a listener for)
- the submit event of the form is fired (you’ve got a function listener for this as well)